### PR TITLE
Add parameter plotting to the plotting script, currently plots histog…

### PR DESCRIPTION
**Pull request by jpa198**
…rams and against magnitude

Changed the Fault and cs v20p4 version file to use the trace length instead of the nhm length to determine magnitude
Fixed the sdrop perturbation bug in the cs v20p4 version file

### CODE REVIEW

File: srf_generation/Fault.py [hunk 1]
```diff
@@ -554,8 +553,19 @@ def __init__(self, nhm_data: NHMFault):
             nhm_data.dip_dir,
         )
 
+        # Circular dependency here
+        # need the magnitude to determine the level of rounding of the length
+        # need the length to determine the magnitude
+        # Shouldn't be an issue in most cases
```
Comment (jasonmotha) [inline srf_generation/Fault.py:559]: Add a note to prospectively move this to the output stage rather than the input stage.

Comment (jpa198) [reply srf_generation/Fault.py:559]: Move length/magnitude calculation to the output stage?

Comment (jpa198) [reply srf_generation/Fault.py:559]: Ie have JIT calculation for magnitude and length?

Comment (jasonmotha) [reply srf_generation/Fault.py:559]: JIT for fault rounding. So it removes the circular dependancy.

Comment (jasonmotha) [reply srf_generation/Fault.py:559]: The length rounding is only to make SRF representation easier.

**Pull request merged**